frontends/ast/ast.cc

```diff
@@ -1457,8 +1457,10 @@ RTLIL::IdString AstModule::derive(RTLIL::Design *design, dict<RTLIL::IdString, R
 
 		// Now that the interfaces have been exploded, we can delete the dummy port related to every interface.
 		for(auto &intf : interfaces) {
-			if(mod->wires_.count(intf.first)) {
-				mod->wires_.erase(intf.first);
+			if(mod->wire(intf.first) != nullptr) {
+				pool<RTLIL::Wire*> to_remove;
+				to_remove.insert(mod->wire(intf.first));
+				mod->remove(to_remove);
```

Comment:
```suggestion
				// Normally, removing wires would be batched together as it's an
				//   expensive operation, however, in this case doing so would mean
				//   that a cell with the same name cannot be created (below)...
				// Since we won't expect many interfaces to exist in a module,
				//   we can let this slide...				
				pool<RTLIL::Wire*> to_remove;
				to_remove.insert(mod->wire(intf.first));
				mod->remove(to_remove);
```